Tomes/TomeArray.cs

```diff
@@ -0,0 +1,335 @@
+		//
+		root = _root;
+		if (root == null) {
+			root = this;
+		}
+
+		//
+		for (int i = 0; i < data.Count; i += 1) {
+			this.Add(Tome.Conjure(data[i], root));
+		}
+		
+		//
+		onChanged += EmitToParents;
+		onAdd += EmitChanged;
+		onDel += EmitChanged;
+	}
+	
+	//
+	private void EmitToParents(JToken oldValue) {
+		if (this != root) {
+			Tome.EmitParentChange(Parent.Parent);
+		}
+	}
+
+	//
+	private void EmitChanged(JToken key) {
+		if (onChanged != null) {
+			onChanged.Invoke(null);
+		}
+	}
+
+
+	//
+	public void Assign(JToken newValue) {
+		switch (newValue.Type) {
+		case JTokenType.Array:
+			TomeArray newTomeArray = new TomeArray((JArray)newValue, root);
+			this.Replace(newTomeArray);
+
+			onChanged -= EmitToParents;
+			onChanged += newTomeArray.onChanged;
+			newTomeArray.onChanged = onChanged;
+			newTomeArray.onDestroy = onDestroy;
+			onAdd -= EmitChanged;
+			onAdd += newTomeArray.onAdd;
+			newTomeArray.onAdd = onAdd;
+			onDel -= EmitChanged;
+			onDel += newTomeArray.onDel;
+			newTomeArray.onDel = onDel;
+
+			if (newTomeArray.onChanged != null) {
+				newTomeArray.onChanged.Invoke(null);
+			}
+			break;
+		case JTokenType.Object:
+			TomeObject newTomeObject = new TomeObject((JObject)newValue, root);
+			this.Replace(newTomeObject);
+
+			onChanged -= EmitToParents;
+			onChanged += newTomeObject.onChanged;
+			newTomeObject.onChanged = onChanged;
+			newTomeObject.onDestroy = onDestroy;
+			onAdd -= EmitChanged;
+			onAdd += newTomeObject.onAdd;
+			newTomeObject.onAdd = onAdd;
+			onDel -= EmitChanged;
+			onDel += newTomeObject.onDel;
+			newTomeObject.onDel = onDel;
+
+			if (newTomeObject.onChanged != null) {
+				newTomeObject.onChanged.Invoke(null);
+			}
+			break;
+		default:
+			TomeValue newTomeValue = new TomeValue((JValue)newValue, root);
+			this.Replace(newTomeValue);
+
+			onChanged -= EmitToParents;
+			onChanged += newTomeValue.onChanged;
+			newTomeValue.onChanged = onChanged;
+			newTomeValue.onDestroy = onDestroy;
+
+			if (newTomeValue.onChanged != null) {
+				newTomeValue.onChanged.Invoke(null);
+			}
+			break;
+		}
+	}
+	
+	//
+	public void Destroy() {
+		if (onDestroy != null) {
+			onDestroy.Invoke();
+		}
+	}
+	
+	//
+	public void Set(int index, JToken value) {
+		// Make sure the property exists, filling in missing indexes
+		if (this.Count <= index) {
+			while (this.Count < index) {
+				this.Add(Tome.Conjure(JValue.CreateNull(), root));
+			}
+
+			this.Add(Tome.Conjure(value, root));
+			if (onAdd != null) {
+				onAdd.Invoke(index);
+			}
+		}
+
+		// Assign the property
+		JToken property = this[index];
+		switch (property.Type) {
+		case JTokenType.Array:
+			(property as TomeArray).Assign(Tome.Conjure(value, root));
+			break;
+		case JTokenType.Object:
+			(property as TomeObject).Assign(Tome.Conjure(value, root));
+			break;
+		default:
+			(property as TomeValue).Assign(Tome.Conjure(value, root));
+			break;
+		}
+	}
+	
+	//
+	public void Del(int index) {
```

Comment:
Object and Array Tomes do have public del function
https://github.com/Wizcorp/node-tomes#del-key-